src/transformStateWithClones.js

```diff
@@ -8,6 +8,34 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const newObj = { ...state };
+  const result = [];
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(newObj, obj.extraData);
+        break;
+
+      case 'removeProperties':
+        for (let i = 0; i < obj.keysToRemove.length; i++) {
+          delete newObj[obj.keysToRemove[i]];
+        }
```

Comment:
i think it would be better to create functions for all cases

Reply:
I don't think so. It takes more place. This code is easy enough to read